zebra-network/src/config.rs

```diff
@@ -46,10 +49,29 @@ impl Config {
     }
 
     /// Get the initial seed peers based on the configured network.
-    pub fn initial_peers(&self) -> HashSet<SocketAddr> {
+    pub async fn initial_peers(&self) -> HashSet<SocketAddr> {
         match self.network {
-            Network::Mainnet => Config::parse_peers(self.initial_mainnet_peers.clone()),
-            Network::Testnet => Config::parse_peers(self.initial_testnet_peers.clone()),
+            Network::Mainnet => Config::parse_peers(self.initial_mainnet_peers.clone()).await,
+            Network::Testnet => Config::parse_peers(self.initial_testnet_peers.clone()).await,
+        }
+    }
+
+    async fn resolve(s: String) -> Option<impl Iterator<Item = SocketAddr>> {
+        // Resolve using `lookup_host`
+        let fut = tokio::net::lookup_host(s.clone());
+        // Add timeout
+        let fut = tokio::time::timeout(Duration::from_nanos(1), fut);
+
+        match fut.await {
+            Ok(Ok(ips)) => Some(ips),
+            Ok(Err(e)) => {
+                tracing::warn!("Can't resolve {:?}. Error: {:?}", s, e);
+                None
+            }
+            Err(e) => {
+                tracing::warn!("Timeout trying to resolve {:?}: {:?}", s, e);
```

Comment:
We try to keep fields separate in `tracing` messages:
https://docs.rs/tracing/0.1.22/tracing/#recording-fields

So that the tracing logs make sense, we need to give fields meaningful names.

Since we have multiple DNS servers, these errors can be logged at info-level.

```suggestion
                tracing::info!(?host, ?e, "DNS error resolving peer IP address");
                HashSet::new()
            }
            Err(e) => {
                tracing::info!(?host, ?e, "DNS timeout resolving peer IP address");
```

Reply:
Thanks. I added this at https://github.com/ZcashFoundation/zebra/pull/1662/commits/8a0c173391102153bee50edfc1ee90e3fa69179e